Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,369 @@
+        return csv_data
+
+
+def rearrange_datetime_data(user_datetime_data, string_format):
+    """
+    Reformat the value of a user field to another format
+
+    :param user_datetime_data: str, the value of the field to be reformatted
+    :param string_format: str, preferred format of value
+    :return:
+    """
+    return datetime.strptime(user_datetime_data, '%Y-%m-%dT%H:%M:%S.%fZ').strftime(string_format)
+
+
+def rearrange_name_data(user_name_title):
+    """
+    Change form of the user's name prefix based on their current value
+
+    :param user_name_title: str, current form of the name prefix
+    :return: str, changed form of the name prefix
+    """
+    match user_name_title:
+        case 'Mrs':
+            return 'missis'
+        case 'Ms':
+            return 'miss'
+        case 'Mr':
+            return 'mister'
+        case 'Madame':
+            return 'mademoiselle'
+
+
+def rearrange_user_time_data(user_timezone_offset):
+    """
+    Calculates the user's time based on their time zone
+
+    :param user_timezone_offset: str, current user timezone offset
+    :return: str, current user time
+    """
+    user_timezone = user_timezone_offset.split(':')
+    hours = int(user_timezone[0][1:]) \
+        if user_timezone[0][0] == '+' or user_timezone[0][0] == '-' \
+        else int(user_timezone[0][0:])
+    minutes = int(user_timezone[1])
+    if user_timezone[0][0] == '-':
+        return (datetime.now(timezone.utc) - timedelta(hours=hours, minutes=minutes)).strftime('%Y-%m-%d %H:%M:%S')
+    elif user_timezone[0][0] == '+':
+        return (datetime.now(timezone.utc) + timedelta(hours=hours, minutes=minutes)).strftime('%Y-%m-%d %H:%M:%S')
+    else:
+        return (datetime.now(timezone.utc)).strftime('%Y-%m-%d %H:%M:%S')
+
+
+def add_new_fields(csv_data):
+    """
+    Add new fields and change others in the data using match cases and datetime libs
+
+    :param csv_data: list, data to which new fields will be added and others will be changed
+    :return: list, data with added and changed values
+    """
+    for i, user_info in enumerate(csv_data, start=1):
+        # ADDING FIELD 'global_index'
+        user_info['global_index'] = i
+
+        # ADDING FIELD 'current_time'
+        user_info['current_time'] = rearrange_user_time_data(user_info['location.timezone.offset'])
+
+        # CHANGING VALUES IN FIELD 'name.title'
+        user_info['name.title'] = rearrange_name_data(user_info['name.title'])
+
+        # CHANGING VALUES IN FIELD 'dob.date'
+        user_info['dob.date'] = rearrange_datetime_data(user_datetime_data=user_info['dob.date'],
+                                                        string_format='%m/%d/%Y')
+
+        # CHANGING VALUES IN FIELD 'register.date'
+        user_info['registered.date'] = rearrange_datetime_data(user_datetime_data=user_info['registered.date'],
+                                                               string_format='%m-%d-%Y, %H:%M:%S')
+    return csv_data
```

Comment:
looks frightening for me.
you already have doc strings in each of these methods. with the comments the code looks massive.
your actions can form one logic block.
please consider such example, but decide by yourself, which version looks better for you
```suggestion
        user_info['global_index'] = i
        user_info['current_time'] = rearrange_user_time_data(user_info['location.timezone.offset'])
        user_info['name.title'] = rearrange_name_data(user_info['name.title'])
        user_info['dob.date'] = rearrange_datetime_data(user_datetime_data=user_info['dob.date'],
                                                        string_format='%m/%d/%Y')
        user_info['registered.date'] = rearrange_datetime_data(user_datetime_data=user_info['registered.date'],
                                                               string_format='%m-%d-%Y, %H:%M:%S')
    return csv_data
```